packages/client/src/pages/Signup/SignUp.tsx

```diff
@@ -0,0 +1,168 @@
+import { useForm } from 'react-hook-form';
+import { Navigate } from 'react-router-dom';
+
+import { FC, useState } from 'react';
+import { authApi } from '../../api/Auth/Auth';
+import { AppLink } from '../../components/AppLink/AppLink';
+import { FormButton } from '../../components/FormButton/FormButton';
+import { Input } from '../../components/Input/Input';
+import { Navbar } from '../../components/Navbar';
+import {
+  validateEmail,
+  validateLogin,
+  validateNames,
+  validatePassword,
+  validatePhone,
+} from '../../helpers/authFormValidation';
+import { isNetworkError } from '../../typeGuards/isNetworkError';
+import { FormValues } from './types';
+
+export const SignUp: FC = () => {
+  const [hasRegistered, setHasRegistered] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    reset,
+    formState: {
+      errors: {
+        root,
+        login: loginError,
+        password: passwordError,
+        firstName: nameError,
+        secondName: surnameError,
+        email: emailError,
+        phone: phoneError,
+      },
+      isSubmitting,
+    },
+  } = useForm<FormValues>({
+    mode: 'onChange',
+  });
+
+  const onSubmit = async ({
+    firstName,
+    secondName,
+    login,
+    email,
+    password,
+    phone,
+  }: FormValues) => {
+    try {
+      await authApi.SignUp({
+        first_name: firstName,
+        second_name: secondName,
+        login,
+        email,
+        password,
+        phone,
+      });
+
+      reset();
+
+      setHasRegistered(true);
+    } catch (error) {
+      let message = 'Что-то пошло не так. Попробуйте перезагрузить страницу';
+
+      if (isNetworkError(error)) {
+        if (error.response.data.reason === 'User already in system') {
+          message = 'Пользователь уже авторизован';
+        }
+
+        if (error.response.data.reason === 'Login already exists') {
+          message = 'Пользователь с таким логином уже существует';
+        }
+
+        if (error.response.data.reason === 'Email already exists') {
+          message = 'Пользователь с такой почтой уже существует';
+        }
+
+        if (error.response.status >= 500) {
+          message = 'Повторите попытку позже';
+        }
+      }
+
+      setError('root', { type: 'server', message });
+    }
+  };
+
+  if (hasRegistered) {
+    return <Navigate to={'/sign-in'} />;
```

Comment:
```suggestion
    return <Navigate to="/sign-in" />;
```
{} нужен чтобы передать какую либо переменную. Обычную строку проще передать напрямую.

Reply:
Рекомендую добавить enum Route:
```ts
enum Route {
   SignIn = 'sign-in'
   Profile = 'profile'
}
```